### DISCUSSION TITLE
Move from Electron to Tauri

### DISCUSSION THREAD

**Discussion by meel-hd** [Future-Scholars/paperlib#482] [Ideas]
I wonder if moving to [Tauri](https://tauri.app) instead of Electron is feasible. Electron is great but it produces huge binaries and it is heavy on RAM and CPU.

Tauri on the other hand is lightweight and produces substantially small binaries in comparison to Electron. That's because Tauri doesn't ship any browser runtime with the app as Electron ships Chromium with each app.

 It's limiting in one case, compatibility with older platforms like Windows 7 that doesn't support the WebView Tauri is based on. Fortunately Tauri fixes that by allowing to build the app with the browser runtime as Electron does.

That said, what do you think about the idea.

**Comment by GeoffreyChen777** [comment]
Hi!

I would say highly impossible.

The reason is that:

1. I know Tauri, it's great, but not good enough. Electron has been verified by various big apps such as discord, VSCode, etc. The community of Electron is much bigger than Tauri's. A good community means it's easy to find solutions to problems.
2. Our extension system highly relies on the `utilityProcess` feature of Electron. It's similar to VSCode's extension system. In Tauri, there is no `utilityProcess`, which means that you need to create a new hidden browser window for every extension. You can imagine how much memory this would consume.
3. Paperlib is not a simple web app. Migrating from Electron to Tauri is not like using Tauri to open a webpage. We have a lot of services need `nodejs` environment such as the core `realm database`.
4. And most of our contributors including me know nothing about rust.

For Electron, the biggest concern is the huge binary. However, very few users care about that. And comparing Paperlib with other paper management tools such as Zotero, I believe that Paperlib is the smallest one.

For CPU and RAM. I think there will be no big difference between Electron and Tauri as they are both based on browsers.
I cannot feel any performance issues on my old intel MacBook with Paperlib.

As you know, there is no browser core in Tauri, which leads to compatibility issues; we cannot guarantee a consistent experience across every operating system's webview.

For the tradeoff between compatibility/development vs binary size, we decide to choose the former.

**Reply by meel-hd** [reply]
I see. The app is deeply integrated with Electron. And Thanks for your quick response.
